DOC Document the file usage table (#9747)
silverstripe/silverstripe-framework: docs/en/02_Developer_Guides/14_Files/07_File_Usage.md

---
title: File usage
summary: See file usage and customising the file "Used on" table
icon: compress-arrows-alt
---

# File Usage

> 

Reviewer: Maybe add a bit of an intro about what the feature does.

```suggestion

CMS users can view where a file is used by accessing the "Used On" tab in the file editor. This feature allows them to identify what DataObjects depend on the File. 

```

Author: Added

In the Files section of the CMS, click on a file to see file details.  Within the file details panel there is a "Used on" tab that shows a table of Pages and other DataObjects where the file is used throughout the website.

## Customising the File "Used on" table in the Files section (asset-admin)

> 

Reviewer: Maybe add a blurb to contextualise this a bit.

```suggestion

Your project specific DataObject will automatically be displayed on the "Used on". This may not always be desirable, especially when working with background DataObjects the user can not interact with directly. Extensions can be applied to the `UsedOnTable` class to suppress specific entries.

```

Author: Added

---
title: File usage
summary: See file usage and customising the file "Used on" table
icon: compress-arrows-alt
---

# File Usage

> CMS users can view where a file is used by accessing the "Used On" tab in the file editor. This feature allows them to identify what DataObjects depend on the File.

Reviewer 2: ```suggestion
CMS users can view where a file is used by accessing the Used on tab in the Files section. This feature allows them to identify what DataObjects depend on the file.
```

Reviewer 2: These suggestions are more in line with the writing structure here: https://userhelp.silverstripe.org/en/4/creating_pages_and_content/creating_and_editing_content/images_and_documents/#file-usage

In the Files section of the CMS, click on a file to see file details.  Within the file details panel there is a

> "Used on" tab that shows a table of Pages and other DataObjects where the file is used throughout the website.

Reviewer 2: ```suggestion
Used on tab that shows a table of Pages and other DataObjects where the file is used throughout the website.
```

## Customising the File "Used on" table in the Files section (asset-admin)

> Your project specific DataObject will automatically be displayed on the "Used on". This may not always be desirable, especially when working with background DataObjects the user can not interact with directly. Extensions can be applied

Reviewer 2: ```suggestion
Your project specific DataObject will automatically be displayed on the Used on tab. This may not always be desirable, especially when working with background DataObjects the user can not interact with directly. Extensions can be applied
```

to the `UsedOnTable` class to update specific entries.

Extension hooks can be used to do the following:
- Exclude DataObjects of a particular type of class from being fetched from the database
- Exclude individual DataObjects that were fetched for showing on the used on table
- Link ancestors of a DataObject so they show on the same row of the used on table

### Example PHP file:

```php
<?php
namespace My\App\Extensions;

use SilverStripe\Core\Extension;
use SilverStripe\ORM\DataObject;

class UsedOnTableExtension extends Extension
{

>     // This extension hook will prevent type(s) of DataObjects from showing on the used on table

Reviewer 2: ```suggestion
    // This extension hook will prevent type(s) of DataObjects from showing on the Used on tab in the Files section
```
